vllm/v1/worker/worker_base.py

```diff
@@ -0,0 +1,125 @@
+# SPDX-License-Identifier: Apache-2.0
+"""Base class for all worker implementations."""
+from abc import ABC, abstractmethod
+from typing import TYPE_CHECKING, Optional
+
+import torch
+import torch.nn as nn
+
+from vllm.config import VllmConfig
+from vllm.logger import init_logger
+from vllm.v1.core.scheduler import SchedulerOutput
+from vllm.v1.kv_cache_interface import KVCacheConfig, KVCacheSpec
+from vllm.v1.outputs import ModelRunnerOutput
+
+if TYPE_CHECKING:
+    from vllm.v1.attention.backends.abstract import AttentionType
+
+logger = init_logger(__name__)
+
+
+class WorkerBase(ABC):
```

Comment:
can we inherit from `vllm.worker.worker_base.WorkerBase` to reduce code duplication, similar to the executor case?

Reply:
hi, just inherit from `vllm.worker.worker_base.WorkerBase`. However, it looks wired since we have to do some fake definitions for abstract methods. Like `add_lora`. Please check.

Reply:
I think you can remove `@abstractmethod` in `vllm.worker.worker_base.WorkerBase` . 

`raise NotImplementedError` there is enough, and then you don't need to add them in v1 worker.